debezium-core/src/main/java/io/debezium/config/CommonConnectorConfig.java

```diff
@@ -85,6 +87,14 @@
             .withDescription("The maximum number of records that should be loaded into memory while performing a snapshot")
             .withValidation(Field::isNonNegativeInteger);
 
+    public static final Field SOURCE_STRUCT_MAKER_CLASS = Field.create("source.struct.make.class")
+            .withDisplayName("Source struct maker")
+            .withType(Type.CLASS)
```

Comment:
I'd prefer if this wouldn't be exposed as a class, rather as an option with the values "default" and "legacy" (the option name then being "source.struct.style" or something like that). We'd resolve these enum values to the corresponding implementations then. The reason being that it gives us more liberty to refactor these things, and also it's easier to use for people.